packages/core/src/codewhispererChat/tools/executeBash.ts

```diff
@@ -0,0 +1,208 @@
+                onStdout: (chunk: string) => {
+                    ExecuteBash.handleChunk(chunk, stdoutBuffer, updates)
+                },
+                onStderr: (chunk: string) => {
+                    ExecuteBash.handleChunk(chunk, stderrBuffer, updates)
+                },
+            }
+
+            const childProcess = new ChildProcess('bash', ['-c', this.command], childProcessOptions)
+
+            try {
+                const result = await childProcess.run()
+                const exitStatus = result.exitCode ?? 0
+                const stdout = stdoutBuffer.join('\n')
+                const stderr = stderrBuffer.join('\n')
+                const [stdoutTrunc, stdoutSuffix] = ExecuteBash.truncateSafelyWithSuffix(
+                    stdout,
+                    maxBashToolResponseSize / 3
+                )
+                const [stderrTrunc, stderrSuffix] = ExecuteBash.truncateSafelyWithSuffix(
+                    stderr,
+                    maxBashToolResponseSize / 3
+                )
+
+                const outputJson = {
+                    exitStatus: exitStatus.toString(),
+                    stdout: stdoutTrunc + (stdoutSuffix ? ' ... truncated' : ''),
+                    stderr: stderrTrunc + (stderrSuffix ? ' ... truncated' : ''),
+                }
+
+                return {
+                    output: {
+                        kind: OutputKind.Json,
+                        content: outputJson,
+                    },
+                }
+            } catch (err: any) {
+                this.logger.error(`Failed to execute bash command '${this.command}': ${err.message}`)
+                throw new Error(`Failed to execute command: ${err.message}`)
+            }
+        })
+    }
+
+    private static handleChunk(chunk: string, buffer: string[], updates: Writable) {
+        const lines = chunk.split(/\r?\n/)
+        for (const line of lines) {
+            updates.write(`${line}\n`)
+            buffer.push(line)
+            if (buffer.length > lineCount) {
+                buffer.shift()
+            }
+        }
+    }
+
+    private static truncateSafelyWithSuffix(str: string, maxLength: number): [string, boolean] {
+        if (str.length > maxLength) {
+            return [str.substring(0, maxLength), true]
+        }
+        return [str, false]
+    }
+
+    private static async whichCommand(cmd: string): Promise<string> {
+        const cp = new ChildProcess('which', [cmd], {
+            collect: true,
+            waitForStreams: true,
+        })
+        const result = await cp.run()
+
+        if (result.exitCode !== 0) {
+            throw new Error(`Command "${cmd}" not found on PATH.`)
+        }
+
+        const output = result.stdout.trim()
+        if (!output) {
+            throw new Error(`Command "${cmd}" found but 'which' returned empty output.`)
+        }
+        return output
+    }
+
+    private static parseCommand(command: string): string[] | undefined {
```

Comment:
Parsing bash/shell commands is non-trivial. Is this intended to support arbitrary bash command, or only very simple cases?